logictest: IndexBackfillMerger span used after finish in high vmodule config

pkg/sql/logictest/tests/fakedist-vec-off/fakedist-vec-off_test.TestLogic_create_as [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/14347062?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/14347062?buildTab=artifacts#/) on master @ [431d56840cacdc86638ccb48e9f3bdc16897d341](https://github.com/cockroachdb/cockroach/commits/431d56840cacdc86638ccb48e9f3bdc16897d341):


```
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 

                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---
panic: use of Span after Finish. Span: IndexBackfillMerger. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

goroutine 24051 [running]:
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0xa941b00?)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:182 +0x114
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc00adca008)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:442 +0x18
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:450
github.com/cockroachdb/cockroach/pkg/util/log.getSpan({0x6fb1c98?, 0xc00a307050?})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:27 +0x34
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x6fb1c98, 0xc00a307050}, 0x2, 0x1, 0x0, 0x60?, {0x59a82f8, 0x1e}, {0xc002535bd8, 0x4, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:104 +0x4b4
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:39
github.com/cockroachdb/cockroach/pkg/util/log.Infof(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:277
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).releaseBytesLocked(0xc00b3df860, {0x6fb1c98, 0xc00a307050}, 0xc60271?)
	github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:1070 +0x295
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).releaseBytes(0xc00559a780?, {0x6fb1c98?, 0xc00a307050?}, 0x41e5a0?)
	github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:1048 +0x9d
github.com/cockroachdb/cockroach/pkg/util/mon.(*BoundAccount).Close(0xc002535d00?, {0x6fb1c98?, 0xc00a307050?})
	github.com/cockroachdb/cockroach/pkg/util/mon/bytes_usage.go:913 +0x3a
github.com/cockroachdb/cockroach/pkg/sql/backfill.(*IndexBackfillMerger).Run.func1()
	github.com/cockroachdb/cockroach/pkg/sql/backfill/mvcc_index_merger.go:114 +0xa6
github.com/cockroachdb/cockroach/pkg/sql/backfill.(*IndexBackfillMerger).Run(0xc00a2a5c08, {0x6fb1c98, 0xc00a306ae0}, {0x6f89a00, 0xc009207508})
	github.com/cockroachdb/cockroach/pkg/sql/backfill/mvcc_index_merger.go:250 +0xcc2
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).StartInternal.func1(0xc00996cf98?)
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:513 +0x5d
created by github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).StartInternal in goroutine 23968
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:512 +0x465
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_create_as.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36575

--- a/pkg/sql/backfill/mvcc_index_merger.go
+++ b/pkg/sql/backfill/mvcc_index_merger.go
@@ -103,6 +103,10 @@ const indexBackfillMergeProgressReportInterval = 10 * time.Second
 
 // Run runs the processor.
 func (ibm *IndexBackfillMerger) Run(ctx context.Context, output execinfra.RowReceiver) {
+	opName := "IndexBackfillMerger"
+	ctx = logtags.AddTag(ctx, opName, int(ibm.spec.Table.ID))
+	ctx, span := execinfra.ProcessorSpan(ctx, ibm.flowCtx, opName, ibm.processorID)
+	defer span.Finish()
 	// This method blocks until all worker goroutines exit, so it's safe to
 	// close memory monitoring infra in defers.
 	mergerMon := execinfra.NewMonitor(ctx, ibm.flowCtx.Cfg.BackfillerMonitor, "index-backfiller-merger-mon")
@@ -113,10 +117,6 @@ func (ibm *IndexBackfillMerger) Run(ctx context.Context, output execinfra.RowRec
 		defer ibm.muBoundAccount.Unlock()
 		ibm.muBoundAccount.boundAccount.Close(ctx)
 	}()
-	opName := "IndexBackfillMerger"
-	ctx = logtags.AddTag(ctx, opName, int(ibm.spec.Table.ID))
-	ctx, span := execinfra.ProcessorSpan(ctx, ibm.flowCtx, opName, ibm.processorID)
-	defer span.Finish()
 	defer output.ProducerDone()
 	defer execinfra.SendTraceData(ctx, ibm.flowCtx, output)
 

